pages/en/cookbook/grafting.mdx

```diff
@@ -0,0 +1,219 @@
+
+## What are Subgraphs?
+
+The Graph allows you to build custom open APIs called "subgraphs". Subgraphs are used to tell indexers (server operators) which data to index on a blockchain and save on their servers in order for you to be able to query it at any time using [GraphQL](https://graphql.org/).
+
+## What is Grafting?
+
+Grafting reuses the data from an existing subgraph and starts indexing it at a later block. This is useful during development to get past simple errors in the mappings quickly or to temporarily get an existing subgraph working again after it has failed
+
+The grafted subgraph can use a GraphQL schema that is not identical to the one of the base subgraph, but merely compatible with it. It has to be a valid subgraph schema in its own right, but may deviate from the base subgraph's schema in the following ways:
+
+- It adds or removes entity types
+- It removes attributes from entity types
+- It adds nullable attributes to entity types
+- It turns non-nullable attributes into nullable attributes
+- It adds values to enums
+- It adds or removes interfaces
+- It changes for which entity types an interface is implemented
+
+For more information, you can check:
+
+- [Grafting](https://thegraph.com/docs/en/developing/creating-a-subgraph/#grafting-onto-existing-subgraphs)
+
+> Note: Grafting requires that the Indexer has indexed the base subgraph. It is not recommended on The Graph Network at this time, and developers should not deploy subgraphs using that functionality to the network via the Studio.
+
+In this tutorial, we will be covering a basic usecase. We will replace an existing contract with an identical contract (with a new address). Then, graft the existing subgraph onto the new contract.
+
+## Building an Existing Subgraph
+
+To be able to build and deploy the existing subgraph used in this tutorial, you need to follow the instructions found here:
+
+1. `@graphprotocol/graph-cli`- This is a command-line tool for building and deploying subgraphs. [Click here](https://www.npmjs.com/package/@graphprotocol/graph-cli) to download using `npm`.
+
+Once these are installed, you can start building the subgraph with the instructions found here:
+
+- [Subgraph example repo](https://github.com/t-proctor/grafting-tutorial)
+
+> Note: The contract was taken from the following [Hackathon Starterkit](https://github.com/schmidsi/hackathon-starterkit).
+
+## Subgraph's Components
+
+There are three components of a subgraph:
+
+### 1. Manifest - `subgraph.yaml`
+
+Defines the data sources of interest, and how they should be processed. The Lock contract is our data source. We will be focusing on this file only in this tutorial.
+
+### 2. Schema - `schema.graphql`
+
+Here you define which data you want to be able to query after indexing your subgraph using GraphQL. This is actually similar to a model for an API, where the model defines the structure of a request body.
```

Comment:
This is technically not correct. The schema simply defines the database schema how you want to store data. The resulting GraphQL schema is generated based on it.

Reply:
Fixed this wording